tools/compiler/src/processor.rs

```diff
@@ -141,11 +144,22 @@ async fn process_chain_config(chain_config: ChainConfig) -> AppResult<Registry>
             .map(Into::into)
             .collect(),
         asset_by_id: all_metadata
+            .clone()
             .into_iter()
             .map(|m| {
-                let id = base64_id(&m)?;
+                let id = base64_id(&m.id())?;
                 Ok((id, m))
             })
             .collect::<AppResult<_>>()?,
+        staking_asset_id: base64_id(&STAKING_TOKEN_ASSET_ID)?,
```

Comment:
Perhaps this is a bad idea and it would be better to pass this through `chain_config`?🤔

Reply:
I wonder, is it possible this to ever change per chain? If not, think it's ok. If so, then we should pass it in.

Reply:
at least in rust code it looks like a constant that does not change depending on chain